packages/logger/tests/unit/middleware/middy.test.ts

```diff
@@ -300,34 +356,69 @@ describe('Middy middleware', () => {
-        });
-        logger.debug('This is a DEBUG log with the user_id');
-        logger.debug('This is another DEBUG log with the user_id');
+      // Act
+      await handler(event, context, () => console.log('Lambda invoked!'));
+
+      // Assess
+      expect(consoleSpy).toBeCalledTimes(2);
+      expect(consoleSpy).toHaveBeenNthCalledWith(1, JSON.stringify({
+        cold_start: true,
+        function_arn: 'arn:aws:lambda:eu-west-1:123456789012:function:foo-bar-function',
+        function_memory_size: 128,
+        function_name: 'foo-bar-function',
+        function_request_id: awsRequestId,
+        level: 'INFO',
+        message: 'Lambda invocation event',
+        service: 'hello-world',
+        timestamp: '2016-06-20T12:08:10.000Z',
+        xray_trace_id: '1-5759e988-bd862e3fe1be46a994272793',
+        event: {
+          foo: 'bar'
+        }
+      }));
+
+    });
+
+    test('when a logger is passed with option logEvent set to false, but POWERTOOLS_LOGGER_LOG_EVENT env var is set to true, it does not log the event', async () => {
+
+      // Prepare
+      process.env.POWERTOOLS_LOGGER_LOG_EVENT = 'true';
+      const logger = new Logger();
+      const consoleSpy = jest.spyOn(logger['console'], 'info').mockImplementation();
+      const lambdaHandler = (): void => {
+        logger.info('This is an INFO log with some context');
+      };
+      const handler = middy(lambdaHandler).use(injectLambdaContext(logger, { logEvent: false }));
+      const event = { foo: 'bar' };
+      const getRandomInt = (): number => Math.floor(Math.random() * 1000000000);
+      const awsRequestId = getRandomInt().toString();
+      const context = {
+        callbackWaitsForEmptyEventLoop: true,
+        functionVersion: '$LATEST',
+        functionName: 'foo-bar-function',
+        memoryLimitInMB: '128',
+        logGroupName: '/aws/lambda/foo-bar-function',
+        logStreamName: '2021/03/09/[$LATEST]abcdef123456abcdef123456abcdef123456',
+        invokedFunctionArn: 'arn:aws:lambda:eu-west-1:123456789012:function:foo-bar-function',
+        awsRequestId: awsRequestId,
+        getRemainingTimeInMillis: () => 1234,
+        done: () => console.log('Done!'),
+        fail: () => console.log('Failed!'),
+        succeed: () => console.log('Succeeded!'),
       };
-      const handler = middy(lambdaHandler).use(injectLambdaContext(logger, { clearState: true }));
-      const persistentAttribs = { ...logger.getPersistentLogAttributes() };
 
       // Act
-      await handler({ user_id: '123456' }, context, () => console.log('Lambda invoked!'));
-      const persistentAttribsAfterInvocation = { ...logger.getPersistentLogAttributes() };
+      await handler(event, context, () => console.log('Lambda invoked!'));
 
       // Assess
-      expect(persistentAttribs).toEqual({
-        foo: 'bar',
-        biz: 'baz'
-      });
-      expect(persistentAttribsAfterInvocation).toEqual(persistentAttribs);
+      expect(consoleSpy).toBeCalledTimes(1);
```

Comment:
Should we also assert here that the event is not really logged in the `consoleSpy`?

Reply:
Added an assertion on the log content to verify that the event is not present.